src/borg/archive.py

```diff
@@ -1074,10 +1074,12 @@ def stat_simple_attrs(self, st, path, fd=None):
             attrs["atime"] = safe_ns(st.st_atime_ns)
         if not self.noctime:
             attrs["ctime"] = safe_ns(st.st_ctime_ns)
-        if not self.nobirthtime and hasattr(st, "st_birthtime"):
-            if is_darwin:
+        if not self.nobirthtime:
+            if hasattr(st, "st_birthtime_ns"):
+                attrs["birthtime"] = safe_ns(st.st_birthtime_ns)
+            elif is_darwin and is_darwin_feature_64_bit_inode:
                 attrs["birthtime"] = safe_ns(get_birthtime_ns(fd or path, follow_symlinks=False))
-            else:
+            elif hasattr(st, "st_birthtime"):
                 # sadly, there's no stat_result.st_birthtime_ns
```

Comment:
guess this comment needs to be removed.

you could add a comment to 1078 noting that this was added in Python 3.12.